### ISSUE TITLE
[Feature Request]  Allow unpiloted mechs to be assigned to TO&E Forces

### ISSUE THREAD

**Issue by Kendricke** [MegaMek/mekhq#307]
As a player/GM, I would like to assign mechs to TO&E whether or not they have pilots for a variety of reasons:
- It helps me to organize units by weight without having to worry about assigning pilots first
- It is another way to track which mechs I have outside of the hanger that require pilots
- Since we can assign technicians to Forces, it is a helpful way to maintain multiple damaged/salvaged units without having to individually assign technicians

**Comment by purpledragon2** [comment]
I can understand where you come from on this one.  I am not one of the developers but I do like good ideas.  However, I am of a different opinion on this.  For example, when I move my force commander from one 'mech to another, I find it anoyying to have to go to the TO&E and remove his old 'mech to then have to put him back in the TO&E in his new 'mech.  

Also, I believe they use the TO&E for purposes of AtB scenario generation.  How are you supposed to keep the OPFOR from generating a overpowering lance because you have a pilotless 'mech assigned to it?

**Comment by kipstafoo** [comment]
I agree, actually (and I am one of the developers :P ). I do this for my tabletop game. Honestly, this would solve one of my annoyances with MHQ -- the constant reminders that I don't have techs on my units I'm ripping apart.

As for the AtB generation comment, that is not a problem. We could create roles on forces. In fact, I've thought about this before. Forces have default roles and for a new contract they are automatically assigned those roles. You can override if you so choose. A force with a role of junkyard/scrap heap/technician's wonderland/etc... would be ignored. Heck, the system could automagically put newly acquired units into this force.

**Comment by neoancient** [comment]
In my mind the TO&E is administrative, and units belongnto certain forces whether or not they are fit for battle. Adjusting for AtB is as simple as filtering out units that are not combat-ready when computing the lance stats.

**Comment by scJazz2** [comment]
The routines for calculating Unit Rating might also need a touch up as well. An empty unit of any type should be ignored by CO or FMMRR rating calculations.

In addition such units shouldn't be added to Contract Payment calculations.

Question: Should units in Mothballs be added as well? If this is purely for organization and eliminating some tedium I would say yes.

**Comment by pheonixstorm** [comment]
Mothballed units included in what exactly? The TOE? I would say no.

**Comment by Kendricke** [comment]
Ran into this issue again and created a workaround (so the TO&E sort of lets you do this already, but it's a pain).  

Basically, I assign a pilot to a mech, pull the mech into the "repair and refit" force, then remove the pilot from that mech.  Now I have a mech with no pilot assigned to a force and I can move it around as I wish.  

So the logic exists on some level for this to happen, it's just a pain to have to work around like this. To work around the workaround, I now have a handful of techs I'm giving Piloting skills to so I can give them a secondary role of Mechwarrior just so I can assign them to mechs and move them into the RR force.
